android/app/src/main/java/app/organicmaps/MwmActivity.java

```diff
@@ -2014,6 +2060,47 @@ public void onRoutingStart()
     RoutingController.get().start();
   }
 
+  public boolean requestUnrestrictedBatteryPermission(UnrestrictedBatteryPermissionRequestedBy requestedBy)
+  {
+    if (!mPowerManager.isPowerSaveMode())
```

Comment:
In some chinese brand phones like xiaomi, huawei, isPowerSaveMode() method always returns false.